### PR TITLE
#163426628 Update user profile

### PR DESCRIPTION
**Pull request by ODINAKACHUKWU**
#### What does this PR do?
Have controller to get, and update a user profile

#### Description of Task to be completed?
- write unit test
- create get, and update user profile controllers

#### How should this be manually tested?
- clone the repo
- create a directory for the project
- cd into the directory
- run `npm install` to install all dependencies
- run `npm run start:dev`
- open post postman and test the controllers using the routes(index.js) in the routes folder

#### What are the relevant pivotal tracker stories?
[#163426628](https://www.pivotaltracker.com/n/projects/2240406/stories/163426628)

### CODE REVIEW

**Review by obasajujoshua31 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -44,13 +44,15 @@
...
     "express": "^4.16.3",
+    "faker": "^4.1.0",
     "joi": "^14.3.1",
     "jsonwebtoken": "^8.4.0",
     "lodash.isempty": "^4.4.0",
     "mailgen": "^2.0.7",
     "methods": "^1.1.2",
     "morgan": "^1.9.1",
-    "passport": "^0.3.0",
+    "multer": "^1.4.1",
```
Comment (benfluleck) [inline package.json]: Remove this please

File: server/models/users.js [hunk 2]
```diff
@@ -50,7 +50,9 @@ export default (sequelize, Sequelize) => {
       type: Sequelize.CHAR
     },
     imageURL: {
-      type: Sequelize.TEXT
+      type: Sequelize.TEXT,
+      defaultValue: `https://lh3.googleusercontent.com/-g4I91oIkL2E
```
Comment (benfluleck) [inline server/models/users.js]: Nooooo go and put it in env variables

File: server/tests/controller/user/profile.spec.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
+import chai from "chai";
+import chaiHttp from "chai-http";
+import app from "../../../index";
+import db from "../../../models";
+import {
+  updateUserData,
+  userData,
```
Comment (benfluleck) [inline server/tests/controller/user/profile.spec.js]: Please use and array

File: server/tests/controller/user/profile.spec.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  await db.sequelize.sync({ force: true });
+});
+
+const invalidId = "bc302642-83e8-4c1b-80b4-c9ed35b6f908";
+const invalidToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ";
+
+describe("Test for User Profile", () => {
+  // let testUserId1;
+  // let testUserId2;
+  // let token1;
```
Comment (benfluleck) [inline server/tests/controller/user/profile.spec.js]: Remove these comments please

**Review by Ndifreke — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -46,11 +46,12 @@
     "express": "^4.16.3",
     "joi": "^14.3.1",
     "jsonwebtoken": "^8.4.0",
+    "lodash-uuid": "0.0.3",
```
Comment (benfluleck) [inline package.json]: Why do we need this library is it a lighter alternative?

File: server/middlewares/findUser/findUser.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import _ from "lodash-uuid";
+import models from "../../models/index";
+import { serverError, httpResponse } from "../../helpers/http";
+
+const { Users } = models;
+
+export const findUserById = async (req, res, next) => {
+  const userId = req.params.id;
+  try {
+    if (!_.isUuid(userId)) {
```
Comment (benfluleck) [inline server/middlewares/findUser/findUser.js]: But you can use Joi to check as well ?

**Review by benfluleck — CHANGES_REQUESTED** [review]
Why did you bring in another library when Joi can do uuid validation?

**Review by obasajujoshua31 — APPROVED** [review]

File: server/controllers/user.js [hunk 7]
```diff
@@ -2,7 +2,7 @@ import isEmpty from "lodash.isempty";
 import { comparePassword } from "../helpers/password";
 import { serverError, httpResponse } from "../helpers/http";
 import { generateJWT, decodeJWT, getJWTConfigs } from "../helpers/jwt";
-import { userResponse } from "../helpers/userResponse";
+import { userResponse, userProfileResponse } from "../helpers/userResponse";
```
Comment (benfluleck) [inline server/controllers/user.js]: I don't like this has two functions userResponse and userProfileResponse in a helper named after one of the functions. Take these 2 functions into the user helpers

File: server/tests/controller/user/profile.spec.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+          lastName: "Drew",
+          gender: "male",
+          biography: "     A son from the sun   ",
+          mobileNumber: 12345678909
+        });
+      expect(res.statusCode).to.equal(400);
+      expect(res.body.success).to.be.false;
+      expect(res.body.errorMessages);
+    });
+    it("should return an error message if user supply invalid token", async () => {
```
Comment (benfluleck) [inline server/tests/controller/user/profile.spec.js:131]: Please put a space between your unit tests for readability

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: server/helpers/passportCallback.js [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { userResponse } from "./userResponse";
+import { userResponse } from "./users";
```
Comment (benfluleck) [inline server/helpers/passportCallback.js]: `userResponses` be consistent with the file you are referencing

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: server/helpers/passportCallback.js (hunk 9 above)
Comment (ghost) [reply server/helpers/passportCallback.js]: I decided to put the `userResponse` and `userProfileResponse` in the same file called `users`, that way if anyone wants to define their own response they can put it there as well...
I thought was what you meant by the initial comment

Comment (ghost) [reply server/helpers/passportCallback.js]: Should I put them in seperate files?

Comment (benfluleck) [reply server/helpers/passportCallback.js]: yea sure, what I meant was change to `userResponses` because there are many of them

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: server/helpers/passportCallback.js (hunk 9 above)
Comment (ghost) [reply server/helpers/passportCallback.js]: Oh, okay... I'll fix it now

**Review by benfluleck — APPROVED** [review]

**Pull request merged**